subscriptions/src/subscription/CachedSubscription.ts

```diff
@@ -11,12 +11,19 @@ export default class CachedSubscription extends Subscription {
         this._cacheKey = cacheKey;
     }
 
+    protected _cachedSubscription = {};
+
+    public async loadCache() {
+        this._cachedSubscription = await this._sdk.cache().getItem(this._cacheKey);
+    }
+
     public subscription() {
-        return this._sdk.cache().getItemSync(this._cacheKey) || {};
+        return this._cachedSubscription || {};
     }
 
     protected _setSubscription(subscription) {
```

Comment:
change this to async?

Reply:
In the basic Subscription module `_setSubscription` is a sync method. If this method were async, this would mean that the basic Subscription related methods would become async. 

But in fact, we just need to make sure to wait for the first async read of the storage subscription, and any subscription modifications to the storage subscription and updates to the in-memory cache subscription can be sync, and do not necessarily need to become async `_setSubscription`, which just writes to storage.